app/controllers/providers/proceedings_types_controller.rb

```diff
@@ -4,40 +4,53 @@ class ProceedingsTypesController < ProviderBaseController
 
     # GET /provider/applications/:legal_aid_application_id/proceedings_types
     def index
+      @multiple_proceedings = allow_multiple_proceedings?
       proceeding_types
     end
 
     # POST /provider/applications/:legal_aid_application_id/proceedings_types
     def create
       return continue_or_draft if draft_selected?
 
-      if legal_aid_application.proceeding_types.present?
+      if run_transaction
         go_forward
       else
         legal_aid_application.errors.add(:'proceeding-search-input', t('.search_and_select'))
+        @multiple_proceedings = allow_multiple_proceedings?
         proceeding_types
         render :index
       end
     end
 
     # PATCH /provider/applications/:legal_aid_application_id/proceedings_types/:id
+    # TODO: Could be removed after multiple proceedings go live.
     def update
-      ActiveRecord::Base.transaction do
-        legal_aid_application.reset_proceeding_types! # This will probably change when multiple proceeding types implemented!
-        legal_aid_application.proceeding_types << proceeding_type
-        AddScopeLimitationService.call(legal_aid_application, :substantive)
-      end
+      run_transaction
       go_forward
     end
 
     private
 
-    def proceeding_type
-      @proceeding_type = ProceedingType.find(params[:id])
+    def run_transaction
+      proceeding_types_service.add(proceeding_type_id: form_params, scope_limitation: :substantive)
+    rescue ActionController::ParameterMissing
+      false
+    end
+
+    def proceeding_types_service
+      ProceedingTypesService.new(legal_aid_application)
+    end
+
+    def form_params
+      params.require(:id)
     end
 
     def proceeding_types
       @proceeding_types ||= ProceedingType.all
     end
+
+    def allow_multiple_proceedings?
+      Setting.allow_multiple_proceedings?
```

Comment:
do `@multiple_proceedings = Setting.allow_multiple_proceedings?` then you don't need to assign twice above